lib/foreman_ansible_core/playbook_runner.rb

```diff
@@ -94,5 +97,41 @@ def initialize_ansible_dir(ansible_dir)
         !ansible_dir.nil? && File.exist?(ansible_dir)
       @ansible_dir = ansible_dir
     end
+
+    def found_in_known_hosts?
```

Comment:
I'd make this `unknown_hosts`:
```ruby
def unknown_hosts
  JSON.parse(@inventory)['all']['hosts'].filter { |host| Net::SSH::Knownhosts.search_for(host).empty? }
end
```

Then you can only scan for hosts you don't already know.

Reply:
In reality the way `foreman_ansible` uses this right now, the 'hosts' key in the inventory only contains one host every time, so it wouldn't make a difference, but I'll update it for the sake of having something solid if we change that.